packages/config-plugins/src/plugins/ios-plugins.ts

```diff
@@ -7,7 +7,10 @@ import type { ExpoPlist, InfoPlist } from '../ios/IosConfig.types';
 import type { AppDelegateProjectFile } from '../ios/Paths';
 import { withMod } from './withMod';
 
-type MutateInfoPlistAction = (expo: ExpoConfig, infoPlist: InfoPlist) => InfoPlist;
+type MutateInfoPlistAction = (
+  expo: ExpoConfig,
+  infoPlist: InfoPlist
+) => Promise<InfoPlist> | InfoPlist;
```

Comment:
correct the type definition for the action function, it supports async but the type said no.